KNeighbor-yigalron.py

```diff
@@ -126,22 +127,48 @@
     clf_grid)
 
 
-# In[13]:
+# In[14]:
 
 get_ipython().run_cell_magic('time', '', '# Fit the model (the computationally intensive part)\npipeline.fit(X=X_train, y=y_train)\nbest_clf = clf_grid.best_estimator_\n#feature_mask = feature_select.get_support()  # Get a boolean array indicating the selected features')
 
 
-# In[16]:
+# In[15]:
 
 print (clf_grid.best_params_)
 
 
-# In[17]:
+# In[22]:
+
+def grid_scores_to_df(grid_scores):
+    """
+    Convert a sklearn.grid_search.GridSearchCV.grid_scores_ attribute to 
+    a tidy pandas DataFrame where each row is a hyperparameter-fold combinatination.
+    """
+    rows = list()
+    for grid_score in grid_scores:
+        for fold, score in enumerate(grid_score.cv_validation_scores):
+            row = grid_score.parameters.copy()
+            row['fold'] = fold
+            row['score'] = score
+            rows.append(row)
+    df = pd.DataFrame(rows)
+    return df
+
+
+# In[24]:
+
+cv_score_df = grid_scores_to_df(clf_grid.grid_scores_)
+facet_grid = sns.factorplot(x='n_neighbors', y='score', col='n_neighbors',
```

Comment:
`n_neighbors` is used as both the x variable and column variable. I would remove `col='n_neighbors'`, which will compact the plot. You could also set `col='weights'` -- however this would only make a difference if you were evaluating multiple options for `weights`, which is not currently the case.

Otherwise, this is a beautiful and very informative visualization.